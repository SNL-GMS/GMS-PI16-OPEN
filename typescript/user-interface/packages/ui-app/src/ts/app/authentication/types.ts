import { UserSessionTypes } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<UserSessionTypes.AuthStatus>;
  checkIsAuthenticated(): Promise<UserSessionTypes.AuthStatus>;
  unAuthenticateWith(): Promise<UserSessionTypes.AuthStatus>;
  logout(setAuthStatus: (status: UserSessionTypes.AuthStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
