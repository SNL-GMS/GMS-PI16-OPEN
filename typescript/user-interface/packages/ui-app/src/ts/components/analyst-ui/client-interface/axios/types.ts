/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  ChannelSegmentTypes,
  EventTypes,
  QcMaskTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import { AxiosRequestConfig } from 'axios';
import { QueryResult } from 'react-query';

export interface EventMutationArgs {
  variables: {
    events: EventTypes.Event[];
  };
}

export interface QcMaskMutationArgs {
  variables: {
    qcMasks: QcMaskTypes.QcMask[];
  };
}

export interface SignalDetectionMutationArgs {
  variables: {
    signalDetections: SignalDetectionTypes.SignalDetection[];
  };
}

// ***************************************
// Workflow Interval Query and Mutation interfaces
// ***************************************
export interface StageIntervalsByStageIdAndTimeRequest {
  startTime: number;
  endTime: number;
  stageIds: WorkflowTypes.WorkflowDefinitionId[];
}
export interface UserRequest {
  userName: string;
  time: number;
}

export interface UpdateActivityIntervalStatusRequest extends UserRequest {
  activityIntervalId: WorkflowTypes.IntervalId;
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface UpdateInteractiveAnalysisStageIntervalStatusRequest extends UserRequest {
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface SplitActivityIntervalRequest {
  intervalId: WorkflowTypes.IntervalId;
  duration: number;
}
export interface EventQueryProps {
  eventQuery: QueryResult<EventTypes.Event[], any>;
}

export interface QcMaskQueryProps {
  qcMaskQuery: QueryResult<QcMaskTypes.QcMask[], any>;
}

export interface SignalDetectionQueryProps {
  signalDetectionQuery: QueryResult<SignalDetectionTypes.SignalDetection[], any>;
}

export interface WaveformQueryProps {
  waveformQuery: QueryResult<ChannelSegmentTypes.ChannelSegment<WaveformTypes.Waveform>[], any>;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: WaveformTypes.WaveformQueryArgs;
};
