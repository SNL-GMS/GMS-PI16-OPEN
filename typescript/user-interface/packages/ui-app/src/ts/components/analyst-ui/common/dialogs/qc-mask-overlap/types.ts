import { QcMaskTypes } from '@gms/common-model';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { CoordinatesPx } from '../qc-mask-form/types';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
