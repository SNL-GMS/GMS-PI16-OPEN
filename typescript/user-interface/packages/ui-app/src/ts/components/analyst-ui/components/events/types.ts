import { GQLTypes } from '@gms/common-graphql';
import { CommonTypes, EventTypes, WorkflowTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import { ChildProps, MutationFunction } from 'react-apollo';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction;
  saveEvent: MutationFunction;
  saveAllModifiedEvents: MutationFunction;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analysisMode: WorkflowTypes.AnalysisMode;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  GQLTypes.EventsInTimeRangeQueryProps &
  GQLTypes.DefaultStationsQueryProps &
  GQLTypes.SignalDetectionsByStationQueryProps &
  GQLTypes.WorkspaceStateProps;
