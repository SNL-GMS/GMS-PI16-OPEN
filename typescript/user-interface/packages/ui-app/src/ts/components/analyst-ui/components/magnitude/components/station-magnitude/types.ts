import { EventTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { MutationFunction } from 'react-apollo';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { AmplitudesByStation } from '../../types';

/**
 * StationMagnitude State
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface StationMagnitudeState {
  computeNetworkMagnitudeSolutionStatus: Immutable.Map<
    string,
    [{ stationName: string; rational: string }]
  >;
}

export interface MagnitudeAndSdData {
  magSolution: EventTypes.NetworkMagnitudeSolution;
  sdData: StationMagnitudeSdData;
}

/**
 * options that can be passed to ag grid
 */
export interface Options {
  alignedGrids: any[];
}

/**
 * StationMagnitude Props
 */
export interface StationMagnitudeProps {
  options?: Options;
  amplitudesByStation: AmplitudesByStation[];
  historicalMode: boolean;
  selectedSdIds: string[];
  locationSolution: EventTypes.LocationSolution;
  displayedMagnitudeTypes: AnalystWorkspaceTypes.DisplayedMagnitudeTypes;
  computeNetworkMagnitudeSolution: MutationFunction;
  openEventId: string;

  setSelectedSdIds(ids: string[]): void;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<[{ stationName: string; rational: string }]>;
}

export interface StationMagnitudeSdData {
  channel: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  signalDetectionId: string;
  time: number;
  stationName: string;
  flagForReview: boolean;
}

export interface MagnitudeDataForRow {
  channel: string;
  signalDetectionId: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  flagForReview: boolean;
  defining: boolean;
  mag: number;
  res: number;
  hasMagnitudeCalculationError: boolean;
  computeNetworkMagnitudeSolutionStatus: string;
}

/**
 * Table row object for station magnitude
 */
export interface StationMagnitudeRow {
  id: string;
  dataForMagnitude: Map<EventTypes.MagnitudeType, MagnitudeDataForRow>;
  station: string;
  dist: number;
  azimuth: number;
  selectedSdIds: string[];
  historicalMode: boolean;
  azimuthTooltip: string;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<void>;
}
export interface StationMagAndSignalDetection {
  stationName: string;
  magnitudeAndSdData: Map<EventTypes.MagnitudeType, MagnitudeAndSdData>;
}
