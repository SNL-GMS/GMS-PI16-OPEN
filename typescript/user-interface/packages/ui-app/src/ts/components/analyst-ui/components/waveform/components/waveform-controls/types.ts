import { CommonTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import React from 'react';

import { QcMaskDisplayFilters } from '~analyst-ui/config';
import { MaskDisplayFilter } from '~analyst-ui/config/user-preferences';

import { AlignWaveformsOn, PanType } from '../../types';
import { AmplitudeScalingOptions, FixedScaleValue } from './scaling-options';

/**
 * Waveform Display Controls Props
 */
export interface WaveformControlsProps {
  defaultSignalDetectionPhase: CommonTypes.PhaseType;
  currentSortType: AnalystWorkspaceTypes.WaveformSortType;
  currentOpenEventId: string;
  analystNumberOfWaveforms: number;
  showPredictedPhases: boolean;
  maskDisplayFilters: QcMaskDisplayFilters;
  alignWaveformsOn: AlignWaveformsOn;
  phaseToAlignOn: CommonTypes.PhaseType | undefined;
  alignablePhases: CommonTypes.PhaseType[] | undefined;
  isMeasureWindowVisible: boolean;
  measurementMode: AnalystWorkspaceTypes.MeasurementMode;
  setMode(mode: AnalystWorkspaceTypes.WaveformDisplayMode): void;
  setDefaultSignalDetectionPhase(phase: CommonTypes.PhaseType): void;
  setSelectedSortType(sortType: AnalystWorkspaceTypes.WaveformSortType): void;
  setAnalystNumberOfWaveforms(value: number, valueAsString?: string): void;
  setMaskDisplayFilters(key: string, maskDisplayFilter: MaskDisplayFilter): void;
  setWaveformAlignment(
    alignWaveformsOn: AlignWaveformsOn,
    phaseToAlignOn: CommonTypes.PhaseType,
    showPredictedPhases: boolean
  ): void;
  toggleMeasureWindow(): void;
  setShowPredictedPhases(showPredicted: boolean): void;
  pan(panDirection: PanType): void;
  onKeyPress(
    e: React.KeyboardEvent<HTMLDivElement>,
    clientX?: number,
    clientY?: number,
    channelId?: string,
    timeSecs?: number
  ): void;
  amplitudeScaleOption: AmplitudeScalingOptions;
  fixedScaleVal: FixedScaleValue;
  setAmplitudeScaleOption: (option: AmplitudeScalingOptions) => void;
  setFixedScaleVal: (val: FixedScaleValue) => void;
}

export interface WaveformDisplayControlsState {
  hasMounted: boolean;
}
