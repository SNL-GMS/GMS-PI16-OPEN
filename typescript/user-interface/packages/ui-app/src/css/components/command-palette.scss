.command-palette-overlay__contents {
  width: 33vw;
  max-width: 40rem;
  padding: 0.5em;
}
.command-palette {
  &__input {
    flex-grow: 1;
    border: 1px solid transparent;
    background: $gms-recessed;
    color: $gms-main;

    & input {
      font-size: 1rem;
    }
  }

  &__results-wrapper {
    max-height: max(50vh, 25em);
    overflow: auto;
    margin: 0 calc(-0.5em + 1px);
  }

  &__result-list {
    padding: 0;
    margin: 0;
    display: flex;
    flex-direction: column;
  }

  &__result {
    flex-grow: 1;
    background: $gms-prominent-background;
    padding: 0.5em;
    margin: 1px 0;
    list-style-type: none;

    &:first-of-type {
      margin-top: 0.5em;
    }

    &:hover {
      @include brighten;
    }

    &--selected {
      background-color: $gms-table-selection;
    }
  }
  & .is-highlighted {
    background-color: rgba($gms-main, 0.15);
  }
}
