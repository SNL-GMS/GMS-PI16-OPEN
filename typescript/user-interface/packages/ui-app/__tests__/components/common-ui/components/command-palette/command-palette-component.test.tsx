import { Client } from '@gms/ui-apollo';
import DefaultClient from 'apollo-boost';
import React from 'react';
import { ApolloProvider } from 'react-apollo';

import { CommandPaletteComponent } from '../../../../../src/ts/components/common-ui/components/command-palette/command-palette-component';
import { CommandPaletteComponentProps } from '../../../../../src/ts/components/common-ui/components/command-palette/types';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

// eslint-disable-next-line import/no-deprecated
const lodash = jest.requireActual('lodash');
lodash.uniqueId = () => '1';

describe('Command Palette Component', () => {
  const client: Client = new DefaultClient<any>();

  it('should be defined', () => {
    expect(CommandPaletteComponent).toBeDefined();
  });

  const props: CommandPaletteComponentProps = {
    commandPaletteIsVisible: true,
    keyPressActionQueue: undefined,
    setCommandPaletteVisibility: jest.fn(),
    setKeyPressActionQueue: jest.fn()
  };

  it('matches a snapshot', () => {
    const wrapper = Enzyme.mount(
      <ApolloProvider client={client}>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <CommandPaletteComponent {...props} />
      </ApolloProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
