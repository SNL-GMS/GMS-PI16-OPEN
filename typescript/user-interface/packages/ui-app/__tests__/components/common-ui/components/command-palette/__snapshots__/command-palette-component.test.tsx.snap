// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command Palette Component matches a snapshot 1`] = `
<ApolloProvider
  client={
    DefaultClient {
      "cache": InMemoryCache {
        "addTypename": true,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": ApolloLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "typeDefs": undefined,
      "version": "2.6.4",
      "watchQuery": [Function],
    }
  }
>
  <CommandPaletteComponent
    commandPaletteIsVisible={true}
    setCommandPaletteVisibility={[MockFunction]}
    setKeyPressActionQueue={[MockFunction]}
  >
    <BaseCommandPaletteOverlay
      commandActions={Array []}
      showCommandPalette={true}
    >
      <OverlayWrapper
        isOpen={true}
        onClose={[Function]}
      >
        <Blueprint3.Overlay
          autoFocus={false}
          backdropClassName=""
          backdropProps={Object {}}
          canEscapeKeyClose={true}
          canOutsideClickClose={true}
          className="overlay"
          enforceFocus={false}
          hasBackdrop={true}
          isOpen={true}
          lazy={true}
          onClose={[Function]}
          transitionDuration={300}
          transitionName="bp3-overlay"
          usePortal={true}
        >
          <Blueprint3.Portal
            container={
              <body
                class="bp3-overlay-open"
              >
                <div
                  class="bp3-portal"
                >
                  <div
                    class="bp3-overlay bp3-overlay-open overlay"
                  >
                    <div
                      class="bp3-overlay-backdrop bp3-overlay-appear bp3-overlay-appear-active"
                      tabindex="0"
                    />
                    <div
                      class="overlay__contents undefined bp3-overlay-content bp3-overlay-appear bp3-overlay-appear-active"
                      tabindex="0"
                    >
                      <div
                        class="overlay__contents command-palette-overlay__contents"
                      >
                        <form
                          class="command-palette"
                          data-cy="command-palette"
                        >
                          <div
                            class="bp3-input-group command-palette__input"
                          >
                            <span
                              class="bp3-icon bp3-icon-chevron-right"
                              icon="chevron-right"
                            >
                              <svg
                                data-icon="chevron-right"
                                height="16"
                                viewBox="0 0 16 16"
                                width="16"
                              >
                                <desc>
                                  chevron-right
                                </desc>
                                <path
                                  d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                            <input
                              autocomplete="off"
                              class="bp3-input"
                              dir="ltr"
                              name="command-palette"
                              placeholder="Command"
                              style="padding-right: 10px;"
                              type="text"
                              value=""
                            />
                          </div>
                          <div
                            class="command-palette__results-wrapper"
                          >
                            <ul
                              class="command-palette__result-list"
                            />
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </body>
            }
          >
            <Portal
              containerInfo={
                <div
                  class="bp3-portal"
                >
                  <div
                    class="bp3-overlay bp3-overlay-open overlay"
                  >
                    <div
                      class="bp3-overlay-backdrop bp3-overlay-appear bp3-overlay-appear-active"
                      tabindex="0"
                    />
                    <div
                      class="overlay__contents undefined bp3-overlay-content bp3-overlay-appear bp3-overlay-appear-active"
                      tabindex="0"
                    >
                      <div
                        class="overlay__contents command-palette-overlay__contents"
                      >
                        <form
                          class="command-palette"
                          data-cy="command-palette"
                        >
                          <div
                            class="bp3-input-group command-palette__input"
                          >
                            <span
                              class="bp3-icon bp3-icon-chevron-right"
                              icon="chevron-right"
                            >
                              <svg
                                data-icon="chevron-right"
                                height="16"
                                viewBox="0 0 16 16"
                                width="16"
                              >
                                <desc>
                                  chevron-right
                                </desc>
                                <path
                                  d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                            <input
                              autocomplete="off"
                              class="bp3-input"
                              dir="ltr"
                              name="command-palette"
                              placeholder="Command"
                              style="padding-right: 10px;"
                              type="text"
                              value=""
                            />
                          </div>
                          <div
                            class="command-palette__results-wrapper"
                          >
                            <ul
                              class="command-palette__result-list"
                            />
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              }
            >
              <TransitionGroup
                appear={true}
                childFactory={[Function]}
                className="bp3-overlay bp3-overlay-open overlay"
                component="div"
                onKeyDown={[Function]}
              >
                <div
                  className="bp3-overlay bp3-overlay-open overlay"
                  onKeyDown={[Function]}
                >
                  <CSSTransition
                    appear={true}
                    classNames="bp3-overlay"
                    in={true}
                    key=".$__backdrop"
                    onExited={[Function]}
                    timeout={300}
                  >
                    <Transition
                      appear={true}
                      enter={true}
                      exit={true}
                      in={true}
                      mountOnEnter={false}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={300}
                      unmountOnExit={false}
                    >
                      <div
                        className="bp3-overlay-backdrop"
                        onMouseDown={[Function]}
                        tabIndex={0}
                      />
                    </Transition>
                  </CSSTransition>
                  <CSSTransition
                    appear={true}
                    classNames="bp3-overlay"
                    in={true}
                    key=".$.0"
                    onExited={[Function]}
                    timeout={300}
                  >
                    <Transition
                      appear={true}
                      enter={true}
                      exit={true}
                      in={true}
                      mountOnEnter={false}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={300}
                      unmountOnExit={false}
                    >
                      <div
                        className="overlay__contents undefined bp3-overlay-content"
                        tabIndex={0}
                      >
                        <CommandPalette
                          commandActions={Array []}
                          defaultSearchTerms={
                            Array [
                              "open display",
                              "close display",
                            ]
                          }
                          isVisible={true}
                        >
                          <CommandPaletteSearchManager
                            commandActions={Array []}
                            defaultSearchResults={Array []}
                            defaultSearchTerms={
                              Array [
                                "open display",
                                "close display",
                              ]
                            }
                            isVisible={true}
                          >
                            <div
                              className="overlay__contents command-palette-overlay__contents"
                              onKeyDown={[Function]}
                            >
                              <form
                                className="command-palette"
                                data-cy="command-palette"
                                onSubmit={[Function]}
                              >
                                <CommandPaletteInput
                                  getSearchTerm={[Function]}
                                  selectionManager={
                                    Object {
                                      "getSelectedResult": [Function],
                                      "resetSelection": [Function],
                                      "selectNext": [Function],
                                      "selectPrevious": [Function],
                                    }
                                  }
                                  setSearchTerm={[Function]}
                                >
                                  <Blueprint3.InputGroup
                                    autoComplete="off"
                                    autoFocus={true}
                                    className="command-palette__input"
                                    dir="ltr"
                                    inputRef={[Function]}
                                    leftIcon="chevron-right"
                                    name="command-palette"
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="Command"
                                    value=""
                                  >
                                    <div
                                      className="bp3-input-group command-palette__input"
                                    >
                                      <Blueprint3.Icon
                                        icon="chevron-right"
                                      >
                                        <span
                                          className="bp3-icon bp3-icon-chevron-right"
                                          icon="chevron-right"
                                        >
                                          <svg
                                            data-icon="chevron-right"
                                            height={16}
                                            viewBox="0 0 16 16"
                                            width={16}
                                          >
                                            <desc>
                                              chevron-right
                                            </desc>
                                            <path
                                              d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                                              fillRule="evenodd"
                                              key="0"
                                            />
                                          </svg>
                                        </span>
                                      </Blueprint3.Icon>
                                      <input
                                        autoComplete="off"
                                        autoFocus={true}
                                        className="bp3-input"
                                        dir="ltr"
                                        name="command-palette"
                                        onChange={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="Command"
                                        style={
                                          Object {
                                            "paddingRight": 10,
                                          }
                                        }
                                        type="text"
                                        value=""
                                      />
                                    </div>
                                  </Blueprint3.InputGroup>
                                </CommandPaletteInput>
                                <CommandPaletteResultList
                                  searchResults={Array []}
                                >
                                  <div
                                    className="command-palette__results-wrapper"
                                  >
                                    <ul
                                      className="command-palette__result-list"
                                    />
                                  </div>
                                </CommandPaletteResultList>
                              </form>
                            </div>
                          </CommandPaletteSearchManager>
                        </CommandPalette>
                      </div>
                    </Transition>
                  </CSSTransition>
                </div>
              </TransitionGroup>
            </Portal>
          </Blueprint3.Portal>
        </Blueprint3.Overlay>
      </OverlayWrapper>
    </BaseCommandPaletteOverlay>
  </CommandPaletteComponent>
</ApolloProvider>
`;
